dspace-api/src/main/java/org/dspace/discovery/indexobject/LDNMessageEntityIndexFactoryImpl.java

```diff
@@ -89,18 +89,19 @@ public SolrInputDocument buildDocument(Context context, IndexableLDNNotification
         doc.addField("queue_status_s", LDNMessageEntity.getQueueStatus(ldnMessage));
         addFacetIndex(doc, "queue_status", String.valueOf(ldnMessage.getQueueStatus()),
             LDNMessageEntity.getQueueStatus(ldnMessage));
-        if (ldnMessage.getObject() == null || ldnMessage.getObject().getID() == null) {
-            throw new SQLException("ldnMessage " + ldnMessage.getID() + " is linked to no Item.");
-        }
-        Item item = itemService.findByIdOrLegacyId(context, ldnMessage.getObject().getID().toString());
-        if (item != null) {
-            addFacetIndex(doc, "object", item.getID().toString(), itemService.getMetadata(item, "dc.title"));
-            addFacetIndex(doc, "relateditem", item.getID().toString(), itemService.getMetadata(item, "dc.title"));
+        if (ldnMessage.getObject() != null || ldnMessage.getObject().getID() != null) {
```

Comment:
@frabacche : I believe this should be an `&&` instead of `||`... otherwise we can still hit NullPointerExceptions if `getID()` is null.